### PR TITLE
Shut down UDP and Packet-Router worker when Gateway stream goes down

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

File: test/hpr_protocol_router_SUITE.erl [hunk 1]
```diff
@@ -231,6 +233,77 @@ server_crash_test(_Config) ->
...
+        end,
+
+    Gateway = hpr_test_gateway:pubkey_bin(GatewayPid),
+    LNS = hpr_route:lns(Route),
+    [{_, #{channel := ChannelPid, stream_pid := StreamPid}}] = ets:lookup(
+        hpr_protocol_router_ets, {Gateway, LNS}
+    ),
+
+    ok = gen_server:stop(GatewayPid),
+    timer:sleep(timer:seconds(2)),
```
Comment (michaeldjeffrey) [inline test/hpr_protocol_router_SUITE.erl:288]: Does the timeout in the receive block not cover this?

File: src/protocols/hpr_protocol_router.erl [hunk 2]
```diff
@@ -98,14 +113,71 @@ get_stream(Gateway, LNS, Server) ->
...
+    #state{} = State
+) ->
+    lager:debug("gateway stream ~p went down: ~p waiting ~wms", [_Pid, _ExitReason, ?CLEANUP_TIME]),
+    erlang:spawn(fun() ->
+        timer:sleep(?CLEANUP_TIME),
+        case hpr_packet_router_service:locate(Gateway) of
+            {error, _Reason} ->
+                lager:debug("connot find a gateway stream: ~p, shutting down", [_Reason]),
+                ?MODULE:remove_stream(Gateway, LNS);
+            {ok, StreamPid} ->
```
Comment (michaeldjeffrey) [inline src/protocols/hpr_protocol_router.erl:165]: monitor the new stream pid, like in gwmp_worker?

File: test/hpr_protocol_router_SUITE.erl (hunk 1 above)
Comment (macpie) [reply test/hpr_protocol_router_SUITE.erl:288]: When a new stream is created it gets monitored every time look at line 116

File: src/protocols/hpr_protocol_router.erl (hunk 2 above)
Comment (macpie) [reply src/protocols/hpr_protocol_router.erl:165]: When a new stream is created it gets monitored every time look at line 116

**Review by michaeldjeffrey — APPROVED** [review]

**Comment by macpie** [comment]
CANNOT BE MERGED  until we know why `hpr_packet_router_downlink_handler` is going down

**Pull request merged**